x-pack/plugins/security_solution/public/notes/api/api.ts

```diff
@@ -6,6 +6,37 @@
  */
 
 import * as uuid from 'uuid';
+import type { BareNote, Note } from '../../../common/api/timeline';
+import { KibanaServices } from '../../common/lib/kibana';
+import { NOTE_URL } from '../../../common/constants';
+
+export const createNote = async ({
+  note,
+  noteId,
+  version,
+  overrideOwner,
+}: {
+  note: BareNote;
+  noteId?: string | null;
+  version?: string | null;
+  overrideOwner?: boolean;
+}) => {
+  let requestBody;
+
+  try {
+    requestBody = JSON.stringify({ noteId, version, note, overrideOwner });
+  } catch (err) {
+    return Promise.reject(new Error(`Failed to stringify query: ${JSON.stringify(err)}`));
```

Comment:
its enough to just throw the error in the async context I think

Reply:
sorry I'm not following. Where would it be enough to throw the error? (by the way, this code is a close duplicate of some existing code, but I'm happy to make it better!)

Reply:
now using `throw`